### PR TITLE
fix: long email address on create Identity

### LINKED ISSUE #4835: BSX: long email address on create Identity
We allow to input longer email address in the filed and therefore it’s showing bug on our site. We should at least specify this to avoid confusion or limit the number of characters

_Originally posted by @cryptodamsky in https://github.com/kodadot/snek/issues/139#issuecomment-1405013216_

### spec
- move `IdentityForm` outside rmrk folder
- use native html verification
- `BasicInput` "max" 32chars prop

### bonus 
- Vue 3 refactoring

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Closes #4835

Details:  
- [x] move IdentityForm outside rmrk folder
- [x] use native html verification
- [x] BasicInput "max" 32chars prop
- [x] Vue 3 refactoring
- [x] fullwidth submit
- [x] fix darkmode issue 

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
mfw I see a form without `<form>` tag

crycat-crying-cat


#### small note
- remove unused code in `IdentityForm`
- could you fix/remove `box` class issue in darkmode
- can we have full width submit button? 🥺

File: components/shared/form/BasicInput.vue [hunk 1]
```diff
@@ -19,26 +19,34 @@
...
+    type: '',
+    required: false,
+    disabled: false,
   }
+)
+
+const input = ref<{ checkHtml5Validity: () => void } | null>(null)
+
+const hasFocus = ref(false)
+const checkValidity = () => {
```
Comment (roiLeo) [inline components/shared/form/BasicInput.vue:48]: is this being used?

File: components/shared/form/BasicInput.vue [hunk 2]
```diff
@@ -19,26 +19,34 @@
...
-  protected hasFocus = false
-  public checkValidity() {
-    return this.input.checkHtml5Validity()
+withDefaults(
+  defineProps<{
+    label: string
+    placeholder: string
+    expanded?: boolean
+    message?: string
+    maxlength?: string
```
Comment (roiLeo) [inline components/shared/form/BasicInput.vue:32]: Warning here `Prop 'message' requires default value to be set`

File: components/common/IdentityForm.vue [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      <BasicInput
+        v-model="identity.legal"
+        :label="$i18n.t('name')"
+        :placeholder="$i18n.t('identity.namePlaceholder')"
+        expanded />
+
+      <BasicInput
+        v-model="identity.email"
+        type="email"
+        :maxlength="32"
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply components/common/IdentityForm.vue]: Updated. I have set the limit only for bsx/snek.

**Comment by Jarsen136** [comment]
> mfw I see a form without `<form>` tag
> 
> 
> #### small note
> * remove unused code in `IdentityForm`
> * could you fix/remove `box` class issue in darkmode
> * can we have full width submit button? 🥺

Thanks. I have updated all of them.

**Review by roiLeo — APPROVED** [review]
✅ code lgtm


on the other hand I just saw that the limit of 32 characters must be set on all fields :/
```log
 Error: createType(Call):: Call: failed decoding identity.setIdentity:: Struct: failed on args: {"info":"{\"additional\":\"Vec<(Data,Data)>\",\"display\":\"Data\",\"legal\":\"Data\",\"web\":\"Data\",\"riot\":\"Data\",\"email\":\"Data\",\"pgpFingerprint\":\"Option<[u8;20]>\",\"image\":\"Data\",\"twitter\":\"Data\"}"}:: Struct: failed on info: {"additional":"Vec<(Data,Data)>","display":"Data","legal":"Data","web":"Data","riot":"Data","email":"Data","pgpFingerprint":"Option<[u8;20]>","image":"Data","twitter":"Data"}:: Struct: failed on display: {"_enum":{"None":"Null","Raw":"Bytes","BlakeTwo256":"H256","Sha256":"H256","Keccak256":"H256","ShaThree256":"H256"}}:: Data.Raw values are limited to a maximum length of 32 bytes
```

**Comment by Jarsen136** [comment]
> on the other hand I just saw that the limit of 32 characters must be set on all fields :/

Good catch! I have updated it.

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 36.79 USD/KSM ~ 1.359 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x3b1343d0c447a89f6a6ac36d02ab28986a8c3da1603554225f472ab1590e5532](https://kusama.subscan.io/extrinsic/0x3b1343d0c447a89f6a6ac36d02ab28986a8c3da1603554225f472ab1590e5532)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**